### PR TITLE
jup program parsers 

### PR DESCRIPTION
**Pull request by Nagaprasadvr**
- [ ]  Parse AMM Programs (ix and accounts) used by Jupiter

### CODE REVIEW

**Review by ray-kast — CHANGES_REQUESTED** [review]
Blocking this on #28.

File: crates/parser/src/jup_programs/raydium/account_parser/parser.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_amm_config_account_parsing() {
+        let parser = RaydiumProgramAccParser;
+
+        let account = account_fixture!("A1BBtTYJd4i3xU8D6Tc2FzU6ZN4oXZWXKZnCxwbHXr8x");
+
+        println!("acc:{:?}", account.clone().account.unwrap().data.len());
```
Comment (kespinola) [inline crates/parser/src/jup_programs/raydium/account_parser/parser.rs]:
```suggestion
```

Capture log stuff in assertions as I imagine you are expecting the same value each run.

File: crates/parser/src/jup_programs/orca/ix_parser/parser.rs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        if let OrcaProgramIx::Swap(accounts, data) = parsed {
+            assert_eq!(
+                accounts.whirlpool.to_string(),
+                "ENYEHSyduTbFN1xoSEGaLu7c1F8AqKucdscMuV5Yypy2".to_string()
+            );
+            assert_eq!(data.a_to_b, false);
+        } else {
+            panic!("Invalid Instruction");
+        }
+    }
```
Comment (kespinola) [inline crates/parser/src/jup_programs/orca/ix_parser/parser.rs]:
```suggestion
        let ix = instruction_fixture!("3WC8LGHHs3wYzWef1YmLsRS96G1s5BV4XJYhzvypgWp1uGG16SxepFCd7FhHaTieW66Yn9JFR4tUPA1HYArgFZaA", 0, 1);

        let parsed = run_ix_parse!(parser, &ix);
        if let OrcaProgramIx::Swap(accounts, data) = parsed {
            assert_eq!(
                accounts.whirlpool.to_string(),
                "ENYEHSyduTbFN1xoSEGaLu7c1F8AqKucdscMuV5Yypy2".to_string()
            );
            assert_eq!(data.a_to_b, false);
        } else {
            panic!("Invalid Instruction");
        }
    }
```

What do you think of this dx? Caller provides the signature and path to the instruction they want. This example is the second inner instruction of the second instruction of the transaction.

We then write just that instruction to the disk at path like `ix-{sig}-0-1`

Comment (Nagaprasadvr) [reply crates/parser/src/jup_programs/orca/ix_parser/parser.rs]: Can be done , but these indexes need to be know before itself , so users need to search for these indexes beforehand

File: crates/parser/src/jup_programs/raydium/account_parser/parser.rs (hunk 1 above)
Comment (Nagaprasadvr) [reply crates/parser/src/jup_programs/raydium/account_parser/parser.rs]: removing this logging which was used for debugging

File: crates/parser/src/jup_programs/raydium/ix_parser/parser.rs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+#[derive(Debug, Clone, Copy)]
+pub struct RaydiumProgramIxParser;
+
+impl Parser for RaydiumProgramIxParser {
+    type Input = InstructionUpdate;
+    type Output = RaydiumProgramIx;
+
+    fn id(&self) -> Cow<str> {
+        "yellowstone_vixen_parser::token_program::RaydiumProgramIxParser".into()
```
Comment (kespinola) [inline crates/parser/src/jup_programs/raydium/ix_parser/parser.rs]:
```suggestion
impl Parser for radium::InstructionParser {
    type Input = InstructionUpdate;
    type Output = RaydiumProgramIx;

    fn id(&self) -> Cow<str> {
        "yellowstone_vixen_parser::token_program::RaydiumProgramIxParser".into()
```

I'm been thinking about mod structure for the parser and think we should expose all pub on the root module in this case `radium` and then name have consistent parser names so `InstructionParser` and `AccountParser`

Comment (kespinola) [reply crates/parser/src/jup_programs/raydium/ix_parser/parser.rs]: Is this ID correct? token_program?

`yellowstone_vixen_parser::jup::radium::InstructionParser`

**Review by kespinola — CHANGES_REQUESTED** [review]
Lets go over the package structure for parsers.

File: crates/parser/src/jup_programs/raydium/ix_parser/parser.rs (hunk 3 above)
Comment (Nagaprasadvr) [reply crates/parser/src/jup_programs/raydium/ix_parser/parser.rs]: will be doing this in a new pr

File: crates/parser/src/token_extension_program/ix_parser/extensions/confidential_transfer_fee.rs [hunk 4]
```diff
@@ -159,7 +159,7 @@ mod proto_parser {
         ConfidentialWithdrawWithheldTokensFromMintAccounts, DisableHarvestToMintAccounts,
         EnableHarvestToMintAccounts, InitializeConfidentialTransferFeeConfigAccounts,
     };
-    use crate::helpers::{FromVecPubkeyToVecString, IntoProto};
+    use crate::helpers::{proto_helpers::FromVecPubkeyToVecString, IntoProto};
```
Comment (kespinola) [inline crates/parser/src/token_extension_program/ix_parser/extensions/confidential_transfer_fee.rs]: If this is already in helpers crate then call this module `proto` `helpers::proto::FromVecPubkeyToVecString`

**Review by kespinola — CHANGES_REQUESTED** [review]

File: crates/parser/src/raydium/account_parser.rs [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+impl ProgramParser for AccountParser {
+    #[inline]
+    fn program_id(&self) -> yellowstone_vixen_core::Pubkey {
+        RADIUM_V3_PROGRAM_ID.to_bytes().into()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use yellowstone_vixen_mock::{account_fixture, run_account_parse, FixtureData};
```
Comment (kespinola) [inline crates/parser/src/raydium/account_parser.rs:94]:
```suggestion
    use yellowstone_vixen_mock::account_fixture;
```

We just need account_fixture now.

File: crates/parser/src/orca/instruction_parser.rs [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+                ))
+            },
+            _ => Err(ParseError::from("Unknown instruction")),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use yellowstone_vixen_mock::{run_ix_parse, tx_fixture, FixtureData};
```
Comment (kespinola) [inline crates/parser/src/orca/instruction_parser.rs]:
```suggestion
    use yellowstone_vixen_mock::tx_fixture;
```

Just need to expose tx_fixture now

**Review by kespinola — CHANGES_REQUESTED** [review]
Just cleanup on the test imports and we are g2g.

**Review by kespinola — APPROVED** [review]

**Pull request merged**